AttendanceSystem.Domain.Model/AttendanceRecord.cs

```diff
@@ -4,5 +4,8 @@ public record AttendanceRecord
 {
     public required SessionId SessionId { get; set; }
     public required StudentId StudentId { get; set; }
-    public required AttendanceKind Record { get; set; }
+    public required AttendanceKind StudentAttendance { get; set; }
+    public required DateTime? StudentSubmitted { get; set; }
+    public required AttendanceKind TeacherAttendance { get; set; }
+    public required DateTime? TeacherSubmitted { get; set; }
```

Comment:
Like @alexanderserraris already mentioned, we could create an `AttendanceSubmittance` (please come up with a better name haha) Value Object (just a record or a struct) that encapsulates the AttendanceKind and the DateTime that it was submitted. Then, we can alter the DbContext and [make those properties 'owned' properties, which mean they will inline in the database.](https://learn.microsoft.com/en-us/ef/core/modeling/owned-entities#configuring-types-as-owned)